src/backend/cdb/cdbpartition.c

```diff
@@ -5261,12 +5272,22 @@ get_part_rule(Relation rel,
 									pid2,
 									bExistError, bMustExist,
 									pSearch, pNode, sid1.data, &pNode2);
+			if (!prule2)
+			{
+				if (!bExistError || !bMustExist)
+					return NULL;
```

Comment:
Why can we return NULL here?

Reply:
The comment in `get_part_rule1()`:
https://github.com/arenadata/gpdb/blob/03e22fbb7e91111e3bff7ceccaba5f7fc04c4d39/src/backend/cdb/cdbpartition.c#L4836-L4840

So if this isn't `AT_AP_IDList` or `AT_AP_IDRule`, and `bExistError` is `false`, we may return `NULL`:
https://github.com/arenadata/gpdb/blob/03e22fbb7e91111e3bff7ceccaba5f7fc04c4d39/src/backend/cdb/cdbpartition.c#L5215-L5220

`get_part_rule()` also may return `NULL` even if `bExistError` is set:
https://github.com/arenadata/gpdb/blob/03e22fbb7e91111e3bff7ceccaba5f7fc04c4d39/src/backend/cdb/cdbpartition.c#L5205-L5206